docs/design/jet/025-jet-job-partition-pruning.md

```diff
@@ -123,9 +123,85 @@ try to choose only the required members.
-`ReadMapOrCacheP` with only required partitions to scan.
+Partition pruning for IMap scan supposed to be a pretty simple optimization: we should extract 
+the partition key condition(s) during SQL opt phase, pass it to a specialized processor meta supplier 
+which will spawn `ReadMapOrCacheP` with only required partitions to scan. We designed this solution with 
+an additional steps to improve solution architecture, which are described below.
+
+#### Implementation details
+First of all, for all prunable `FullScan`-s in resulting plan we want to have a **precise** partition set to scan. 
+To make it possible, and also isolate the implementation for each specific connector, we move the computational process
+to `SqlConnector`. We extended `fullScanReader` method in `SqlConnector` interface to accept extracted all 
+partition pruning candidates as a parameter and calculate it in connector-specific way:
+```
+    @Override
+    @Nonnull
+    public Vertex fullScanReader(
+            @Nonnull DagBuildContext context,
+            @Nullable HazelcastRexNode predicate,
+            @Nonnull List<HazelcastRexNode> projection,
+            @Nullable List<Map<String, Expression<?>>> partitionPruningCandidates, // <-- new parameter
+            @Nullable FunctionEx<ExpressionEvalContext, EventTimePolicy<JetSqlRow>> eventTimePolicyProvider)
+```
+
+The new parameter `partitionPruningCandidates` is a list of maps, where for each column present in the filter 
+the column name maps to the extracted comparison expression. The connector-specific implementation should
+filter predicates for columns where partitioning strategy is applicable, and check if partitioning strategy 
+is applicable generally, and in case of success transform the input to the **list of inner list of expressions**. 
+Each inner list of expressions contains comparison expressions, and may be **one-sized**, 
+if the partitioning strategy key is **simple**, and **multi-sized**, if the partitioning strategy key is **composite**. 
+If we have more than one prunable filter predicate, outer list will be multi-sized. 
+
+**Currently, it is implemented only for IMap connector, where all expressions are available.**
+
+For better imagination we prepared an example below.
+
+#### Successful case example 
+Let's assume we have an IMap `map` with composite key {comp1, comp2, comp3} and applied partitioning strategy for
+`comp1` and `comp2`. Let's have the following synthetic query, where filter matches the partitioning strategy:
+```
+SELECT * FROM map WHERE __key.comp1 = 1 AND __key.comp2 = 2
+```
+IMap-specific `fullScanReader` receives the following list of maps as a parameter:
+```
+[{"__key.comp1" = Expression(`__key.comp1 = 1`)}, {"__key.comp2" = Expression(`__key.comp2 = 2`)}]
+```
+
+After the described computation above, `fullScanReader` implementation should return the following list of expressions:
```

Comment:
"`fullScanReader` implementation should *return*"? rather pass to PMS/vertex or something like that.